src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -355,6 +363,23 @@ else if (isArray()) {
             throw new IllegalStateException(descriptorString());
     }
 
+    /**
+     * {@return the string representation of this {@linkplain ClassDesc} in a
+     * {@link ##constant-class-info CONSTANT_Class_info} structure}
+     *
+     * @apiNote
+     * In a future release, this API may return a value instead of throwing
+     * an exception if this {@linkplain ClassDesc} becomes representable in a
+     * {@code CONSTANT_Class_info}.
+     *
+     * @throws IllegalStateException if this {@linkplain ClassDesc} describes a type
```

Comment:
This should really throw `UnsupportedOperationException`, as the instances are immutable, same as [`Class::arrayType()`]. See [JDK‑8268250] and [GH‑4382] for discussion.

[`Class::arrayType()`]: https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/Class.html#arrayType()
[JDK‑8268250]: https://bugs.openjdk.org/browse/JDK-8268250
[GH‑4382]: https://github.com/openjdk/jdk/pull/4382

Reply:
The choice is based on ClassDesc.nested